WalletWasabi/Wallets/Kitchen.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Crypto;
+using WalletWasabi.Crypto.Randomness;
+
+namespace WalletWasabi.Wallets
+{
+	public class Kitchen
+	{
+		private string? Salt { get; set; } = null;
+		private string? Soup { get; set; } = null;
+		private object RefrigeratorLock { get; } = new object();
+
+		[MemberNotNullWhen(returnValue: true, nameof(Salt), nameof(Soup))]
```

Comment:
@yahiheb this property attribute here denotes the code analyzer that if this returns true then Salt and Soup cannot be null. I think there are many cases in the code where we are having a property for the null check but this attribute is missing. Can you take a look around and add this attribute, this will reduce the number of nullable warnings.